### PR TITLE
Fix Qt Creator code model parsing by quoting *_EXPORT undefines

### PR DESCRIPTION
**Pull request by dakcarto**
When parsing, the REMOVE_DEFINITIONS is acknowledged, but not the extra-quoted ADD_DEFINITIONS when switching from DLLIMPORT to DLLEXPORT. This caused Qt Creator’s code completion to fail miserably.

This extra quoting on the `-U` statement essentially makes both the removal and re-definition of the *_EXPORT definitions invisible to Creator's parsing utility, while still remaining viable to the compiler, albeit adding verbosity to the compile command string.

At least with Apple Clang, the extra quoting is appropriate for the re-definition to DLLEXPORT; i.e., while removing the extra quoting on it _does_ fix the code model parsing, it then fails to compile due to (appropriate) spaces in the replaced DLLEXPORT text.

**NOTE: I am unsure if this swap of technique for removing a definition will be recognized on Windows.** @jef-n, any insights here?

Another variation might be to keep the current functionality for Windows, e.g.:
```
IF (NOT MSVC)
  ADD_DEFINITIONS("\"-UCORE_EXPORT\"")
ELSE (NOT MSVC)
  REMOVE_DEFINITIONS("-DCORE_EXPORT=${DLLIMPORT}")
ENDIF (NOT MSVC)
ADD_DEFINITIONS("\"-DCORE_EXPORT=${DLLEXPORT}\"")
```

or, conditionally based upon the compiler.

### CODE REVIEW

**Comment by dakcarto** [comment]
Previous mailing list posting, for reference: http://osgeo-org.1560.x6.nabble.com/DLL-import-export-defines-quoting-td5288826.html

File: src/analysis/CMakeLists.txt [hunk 1]
```diff
@@ -172,7 +172,7 @@ INCLUDE_DIRECTORIES(SYSTEM
   ${SQLITE3_INCLUDE_DIR}
 )
 
-REMOVE_DEFINITIONS("-DANALYSIS_EXPORT=${DLLIMPORT}")
+ADD_DEFINITIONS("\"-UANALYSIS_EXPORT\"")
```
Comment (nyalldawson) [inline src/analysis/CMakeLists.txt]: I don't think -U should be used with ADD_DEFINITIONS. See http://cmake.3232098.n2.nabble.com/undef-a-symbol-td7580326.html

**Comment by m-kuhn** [comment]
I think the proper solution would be [GenerateExportHeaders](https://cmake.org/cmake/help/v3.0/module/GenerateExportHeader.html), while trivial in theory, the hard part is including the additional file everywhere

**Comment by m-kuhn** [comment]
See https://github.com/qgis/QGIS/pull/3944

**Comment by m-kuhn** [comment]
closing this as superseded by #3944

**Pull request closed without merging**